src/test/java/com/example/weather/client/mappers/WeatherDataMapperTest.java

```diff
@@ -0,0 +1,54 @@
+package com.example.weather.client.mappers;
+
+import com.example.weather.client.models.dto.WeatherDataDto;
+import com.example.weather.client.models.dto.WeatherDto;
+import com.example.weather.client.models.entity.Weather;
+import com.example.weather.client.models.entity.WeatherData;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.co.jemos.podam.api.PodamFactory;
+import uk.co.jemos.podam.api.PodamFactoryImpl;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+@ExtendWith(MockitoExtension.class)
+class WeatherDataMapperTest {
+
+    private static PodamFactory FACTORY = new PodamFactoryImpl();
+    private static final WeatherDataDto WEATHER_DATA_DTO = FACTORY.manufacturePojo(WeatherDataDto.class);
+
+    private GeoCoordinatesMapper geoCoordinatesMapper = new GeoCoordinatesMapper();
+    private WeatherMapper weatherMapper = new WeatherMapper();
+    private ConditionsMapper conditionsMapper = new ConditionsMapper();
+    private WindMapper windMapper = new WindMapper();
+    private WeatherDataMapper weatherDataMapper =
+            new WeatherDataMapper(geoCoordinatesMapper, weatherMapper, conditionsMapper, windMapper);
+
+    @Test
+    void testToEntity() {
+        var weatherData = weatherDataMapper.toEntity(WEATHER_DATA_DTO);
+        assertThat(weatherData).isNotNull();
+        assertEqualsForWeatherData(weatherData);
+    }
+
+    private void assertEqualsForWeatherData(WeatherData weatherData) {
+        assertEquals(weatherData.getCoordinates().getLongitude(), WEATHER_DATA_DTO.getCoord().getLon());
```

Comment:
In assertEquals first argument is what is expected, and second is actual value.
Either swap the arguments or use assertJ